### PR TITLE
Update ChiselRunners to use svsim

### PR DESCRIPTION
**Pull request by GeorgeLyon**
This PR moves `BasicTester` style tests over to using `svsim`.

#### API Impact

I tweaked the `simulator` API to make it easier to use without a `Simulator` (i.e. make the code which glues Chisel modules to `svsim.Workspace` more modular). This API has not yet been released so we can change it freely until Chisel 5. 

#### Release Notes

- All `ChiselRunner` tests now use `svsim` when calling `assertTester{Passes,Fails}`

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x` or `3.6.x` depending on impact, API modification or big change: `5.0.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/test/scala/chiselTests/ChiselSpec.scala
Comment (GeorgeLyon) [inline src/test/scala/chiselTests/ChiselSpec.scala:109]: This isn't ideal, and we eventually want to have more structured detection of assertions and `$finish`. I leave it up to the maintainers whether or not we commit this hack now or wait for better finish/assertion support.

File: .devcontainer/devcontainer.json
Comment (chick) [inline .devcontainer/devcontainer.json:15]: Missing new line at end of file

**Review by chick — APPROVED** [review]
I did not fully internalize but it looks good as it seems to be a fairly straightforward refactoring. Github complains about missing newline at end of devcontrainer.json, I leave that to you as to whether it should be fixed.

**Comment by GeorgeLyon** [comment]
@sifive-bot start-ci

**Pull request merged**
